packages/big-design/src/components/Stepper/Stepper.tsx

```diff
@@ -0,0 +1,104 @@
+import { CheckIcon } from '@bigcommerce/big-design-icons';
+import React, { HTMLAttributes, memo, useEffect, useState } from 'react';
+
+import {
+  StyledDash,
+  StyledLight,
+  StyledLightText,
+  StyledMobileLight,
+  StyledMobileLightText,
+  StyledStep,
+  StyledStepper,
+  StyledText,
+} from './styled';
+
+export interface StepProps {
+  number: number;
+  text?: string;
+  state?: 'pending' | 'current' | 'done';
+  isLast?: boolean;
+}
+
+export interface StepperProps extends HTMLAttributes<HTMLDivElement> {
+  steps: Array<string>;
+  currentStep: number;
+}
+
+export interface LightProps {
+  active?: boolean;
+}
+export interface DashProps {
+  active?: boolean;
+}
+export interface TextProps {
+  active?: boolean;
+}
+
+export const Step: React.FC<StepProps> = ({ number, text, state, isLast }) => {
+  const [isCurrent, setIsCurrent] = useState(state === 'current');
+  const [isDone, setIsDone] = useState(state === 'done');
+  useEffect(() => {
+    setIsCurrent(state === 'current');
+    setIsDone(state === 'done');
+  }, [state]);
+
+  return (
+    <StyledStep>
+      <StyledLight active={isCurrent || isDone}>
+        {isDone ? (
+          <CheckIcon color="white" size="large" style={{ padding: '2px' }} />
+        ) : (
+          <StyledLightText>{number}</StyledLightText>
+        )}
+      </StyledLight>
+      <StyledDash active={isDone} hidden={isLast} />
+      <StyledText active={isCurrent}>{text}</StyledText>
+    </StyledStep>
+  );
+};
+
+export const Stepper: React.FC<StepperProps> = memo(({ className, style, ...props }) => {
+  const [currentStep, setCurrentStep] = useState(0);
+  useEffect(() => {
+    if (props.currentStep >= 0 && props.currentStep < props.steps.length) setCurrentStep(props.currentStep);
+  }, [props.currentStep]);
+
+  const determineStepState = (index: number) => {
+    if (currentStep === index) return 'current';
+    else if (currentStep < index) return 'pending';
+    else return 'done';
+  };
+
+  return (
+    <StyledStepper>
+      <StyledMobileLight>
```

Comment:
See my branch on how I was able to use mobile stylings within the `Step` component to avoid using an extra component.